jina/executors/indexers/keyvalue.py

```diff
@@ -23,8 +23,20 @@ class _WriteHandler:
     """
 
     def __init__(self, path, mode):
-        self.body = open(path, mode)
-        self.header = open(path + '.head', mode)
+        self.path = path
+        self.mode = mode
+        self.body = open(self.path, self.mode)
+        self.header = open(self.path + '.head', self.mode)
+
+    def __enter__(self):
+        if self.body.closed:
+            self.body = open(self.path, self.mode)
+        if self.header.closed:
+            self.header = open(self.path + '.head', self.mode)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.flush()
```

Comment:
Why don't we `close()` on exit?

Reply:
Because we wanted to avoid the need to reopen all the time, but at least we assume the data is flushed

Reply:
Closing the file after every 'add' will be quite expensive.

Reply:
Yes, got it. That's what I tried in the previous approach. And we're kinda doing "half" of the ctx management. Makes sense